tests/scaletest/scale_test.go

```diff
@@ -391,78 +574,285 @@ func DeleteIngressesSerial(t *testing.T, numOfIng int, initialNumOfPools int, Av
 	t.Logf("Deleted %d Pools", numOfIng)
 }
 
-func SerialIngressHelper(t *testing.T, numOfIng int) {
-	pools := lib.FetchPools(t, AviClients[0])
-	initialNumOfPools = len(pools)
-	VSes := lib.FetchVirtualServices(t, AviClients[0])
-	initialNumOfVSes = len(VSes)
-	CreateIngressesSerial(t, numOfIng, initialNumOfPools)
-	DeleteIngressesSerial(t, numOfIng, initialNumOfPools, AviClients[0])
-	t.Logf("%d %s Ingress serially creation deletion along with verification is done", numOfIng, ingressType)
+func HybridCreation(t *testing.T, wg *sync.WaitGroup, numOfIng int, deletionStartPoint int) {
+	for i := deletionStartPoint; i < numOfIng; i++ {
+		mutex.Lock()
+		wg.Add(1)
+		var ingresses []string
+		var err error
+		switch {
+		case ingressType == INSECURE:
+			ingresses, _, err = lib.CreateInsecureIngress(ingressNamePrefix, listOfServicesCreated[0], namespace, 1, i)
+			if err != nil {
+				t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+			}
+		case ingressType == SECURE:
+			ingresses, _, err = lib.CreateSecureIngress(ingressNamePrefix, listOfServicesCreated[0], namespace, 1, i)
+			if err != nil {
+				t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+			}
+		case ingressType == MULTIHOST:
+			ingresses, _, _, err = lib.CreateMultiHostIngress(ingressNamePrefix, listOfServicesCreated, namespace, 1, i)
+			if err != nil {
+				t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+			}
+		}
+		t.Logf("Created ingresses %s", ingresses)
+		ingressesCreated = append(ingressesCreated, ingresses...)
+		time.Sleep(time.Second * 1)
+		mutex.Unlock()
+		defer wg.Done()
+
+	}
+	defer wg.Done()
+}
+
+func HybridUpdation(t *testing.T, wg *sync.WaitGroup, numOfIng int) {
+	for len(ingressesUpdated) < numOfIng {
+		mutex.Lock()
+		wg.Add(1)
+		tempStr := DiffOfLists(ingressesCreated, ingressesDeleted)
+		toUpdateIngresses := DiffOfListsOrderBased(ingressesUpdated, tempStr)
+		if len(toUpdateIngresses) > 0 {
+			updatedIngresses, err := lib.UpdateIngress(namespace, toUpdateIngresses)
+			if err != nil {
+				t.Fatalf("Error updating ingresses as : %v ", err)
+				return
+			}
+			t.Logf("Updated ingresses %s", updatedIngresses)
+			ingressesUpdated = append(ingressesUpdated, updatedIngresses...)
+		}
+		time.Sleep(time.Second * 1)
```

Comment:
Check if we can use a periodic timer and a channel, e.g. 
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L420

Reply:
Eliminating the wait part altogether